server-embedded/src/main/java/org/exoplatform/chat/server/ChatServer.java

```diff
@@ -113,7 +114,23 @@ public Response.Content index() throws IOException
   {
     return index.ok();
   }
-
+  
+  @Resource
+  @Route("/filterOutSilentUsers")
+  public Response.Content getFilteredList(String user,List<String> roomUsers, String roomId, String token){
```

Comment:
Couldn't we calculate the list of receivers without getting them from the Front layer using UserService ?

Reply:
this is the solution i was going to implement but i found problems injecting the NotificationService object to userRestService